pkg/stanza/fileconsumer/matcher/matcher.go

```diff
@@ -109,20 +110,24 @@ type Matcher struct {
 
 // MatchFiles gets a list of paths given an array of glob patterns to include and exclude
 func (m Matcher) MatchFiles() ([]string, error) {
-	files := finder.FindFiles(m.include, m.exclude)
+	var errs error
+	files, err := finder.FindFiles(m.include, m.exclude)
+	if err != nil {
+		errs = errors.Join(errs, err)
+	}
 	if len(files) == 0 {
-		return files, fmt.Errorf("no files match the configured criteria")
+		return files, errors.Join(errs, fmt.Errorf("no files match the configured criteria"))
```

Comment:
Nit: Probably makes more sense to start with the high level err and then show detailed errors.
```suggestion
		return files, errors.Join(fmt.Errorf("no files match the configured criteria"), errs)
```